src/test/java/ir/sahab/kafkaconsumer/OffsetTrackerTest.java

```diff
@@ -127,6 +127,31 @@ public void testPageWithTailGap() {
         Assert.assertFalse(offsetToCommit.isPresent());
     }
 
+    @Test
+    public void testPageGap() {
+        final int pageSize = 3;
+        final int maxOpenPagesPerPartition = 4;
+        final OffsetTracker offsetTracker = new OffsetTracker(pageSize, maxOpenPagesPerPartition, new MetricRegistry());
+        final int partition = 0;
+
+        // Track calls which opens the first page: [0..2]
+        offsetTracker.track(partition, 1);
+
+        // Track calls which opens the second page: [9..11] and makes a gap for the first page
+        // Offset 1 completes the first page because the next offsets are inside the recognized gap.
```

Comment:
```suggestion
        // Offset 1 completes the first page and the other pages located in the gap.
```
Please also move this comment and put it for the next block.